### PR TITLE
Port funcall-interactively, replace Ffuncall with funcall

### PR DESCRIPTION
**Pull request by nickdrozd**

### CODE REVIEW

File: rust_src/src/callint.rs [hunk 1]
```diff
@@ -74,4 +80,17 @@ use remacs_macros::lisp_fn;
 #[lisp_fn(min = "0", unevalled = "true")]
 pub fn interactive(_args: LispObject) {}
```
Comment (shaleh) [inline rust_src/src/callint.rs]: maybe `from compiled code` instead of `from C`? Or `from outside of lisp`?

File: rust_src/src/callint.rs [hunk 2]
```diff
@@ -74,4 +80,17 @@ use remacs_macros::lisp_fn;
...
 
+/* BEWARE: Calling this directly from C would defeat the purpose!  */
+/// Like `funcall' but marks the call as interactive.
+/// I.e. arrange that within the called function `called-interactively-p' will return non-nil.
+/// usage: (funcall-interactively FUNCTION &rest ARGUMENTS)
```
Comment (shaleh) [inline rust_src/src/callint.rs]: This is missing `min="1"`.

File: rust_src/src/callint.rs [hunk 3]
```diff
@@ -74,4 +80,17 @@ use remacs_macros::lisp_fn;
...
+#[lisp_fn]
+pub fn funcall_interactively(args: &mut [LispObject]) -> LispObject {
+    let count = c_specpdl_index();
+
+    unsafe { temporarily_switch_to_single_kboard(std::ptr::null_mut() as *mut Lisp_Frame) };
```
Comment (brotzeit) [inline rust_src/src/callint.rs]: I think the cast isn't necessary.

File: rust_src/src/callint.rs [hunk 4]
```diff
@@ -74,4 +80,17 @@ use remacs_macros::lisp_fn;
...
+
+    unsafe { unbind_to(count, funcall(args)) }
```
Comment (brotzeit) [inline rust_src/src/callint.rs]: `unbind_to` returns `nil`. No need to return `LispObject`.

File: rust_src/src/callint.rs [hunk 5]
```diff
@@ -74,4 +80,17 @@ use remacs_macros::lisp_fn;
 #[lisp_fn(min = "0", unevalled = "true")]
 pub fn interactive(_args: LispObject) {}
 
+/* BEWARE: Calling this directly from C would defeat the purpose!  */
+/// Like `funcall' but marks the call as interactive.
+/// I.e. arrange that within the called function `called-interactively-p' will return non-nil.
+/// usage: (funcall-interactively FUNCTION &rest ARGUMENTS)
+#[lisp_fn]
```
Comment (brotzeit) [inline rust_src/src/callint.rs]: The function has a minimum of 1 "1, MANY, 0".

File: rust_src/src/callint.rs (hunk 1 above)
Comment (nickdrozd) [reply rust_src/src/callint.rs]: fixed

File: rust_src/src/callint.rs (hunk 3 above)
Comment (nickdrozd) [reply rust_src/src/callint.rs]: fixed

File: rust_src/src/callint.rs (hunk 2 above)
Comment (nickdrozd) [reply rust_src/src/callint.rs]: fixed

**Comment by nickdrozd** [comment]
I noticed that a few other functions were also missing `min = "1"`, so I added those in as well as a few test cases. It turns out that executing `(funcall)` will crash Remacs!

**Comment by shaleh** [comment]
remacs master crashes without your PR.

**Comment by shaleh** [comment]
Here is some lldb from my machine. I opened up `ielm` and ran `(funcall)`.

```
(lldb) up   # This is the beginning of the panic unwind.
frame #13: 0x0000000100699584 remacs`core::panicking::panic_bounds_check::h2ed47401a769aabb at panicking.rs:59 [opt]
(lldb) up
frame #14: 0x0000000100466b54 remacs`Ffuncall + 772
remacs`Ffuncall:
    0x100466b54 <+772>: leaq   0x2af8fd(%rip), %rdi      ; str.0 + 313
    0x100466b5b <+779>: movl   $0x2a, %esi
    0x100466b60 <+784>: callq  0x1002658e0               ; make_string at alloc.c:2433
    0x100466b65 <+789>: xorl   %esi, %esi
(lldb) up
frame #15: 0x00000001002af2aa remacs`eval_sub(form=4556745427) at eval.c:1493
   1490
   1491		  set_backtrace_args (specpdl + count, vals, argnum);
   1492
-> 1493		  val = XSUBR (fun)->function.aMANY (argnum, vals);
   1494
   1495		  check_cons_list ();
   1496		  lisp_eval_depth--;
(lldb) p XSUBR(fun)
(Lisp_Subr *) $3 = 0x0000000103306848
(lldb) p XSUBR(fun)->function
(Lisp_Subr::(anonymous union)) $4 = {
  a0 = 0x0000000100466850 (remacs`Ffuncall)
  a1 = 0x0000000100466850 (remacs`Ffuncall)
  a2 = 0x0000000100466850 (remacs`Ffuncall)
  a3 = 0x0000000100466850 (remacs`Ffuncall)
  a4 = 0x0000000100466850 (remacs`Ffuncall)
  a5 = 0x0000000100466850 (remacs`Ffuncall)
  a6 = 0x0000000100466850 (remacs`Ffuncall)
  a7 = 0x0000000100466850 (remacs`Ffuncall)
  a8 = 0x0000000100466850 (remacs`Ffuncall)
  aUNEVALLED = 0x0000000100466850 (remacs`Ffuncall)
  aMANY = 0x0000000100466850 (remacs`Ffuncall)
}
```

`argnum` is zero as expected.

On my copy of Emacs this emits "wrong number of arguments" as expected.

**Comment by shaleh** [comment]
And adding the `min="1"` fixes it. Sweet.

**Pull request merged**